csrc/scheduler/matmul_heuristic.h

```diff
@@ -193,7 +193,32 @@ class MatmulParams : public HeuristicParams {
 
   //! This is the CGA size on Hopper+ devices. This parameter is ignored on
   //! Ampere and Turing.
-  std::tuple<int64_t, int64_t, int64_t> cluster_dims = {1, 1, 1};
+  struct ClusterDims {
+    int64_t x = 1;
+    int64_t y = 1;
+    int64_t z = 1;
+
+    bool operator==(const ClusterDims& other) const {
+      return x == other.x && y == other.y && z == other.z;
+    }
+
+    bool operator!=(const ClusterDims& other) const {
+      return !(*this == other);
+    }
+
+    std::string toString() const {
+      std::stringstream ss;
+      ss << x << " " << y << " " << z;
```

Comment:
Print cluster dimensions in its cuda cpp format for clarity
```suggestion
      ss << "__cluster_dims__(" << x << ", " << "Y" << ", " << "Z" << ")";
```

Reply:
Alternative:
```suggestion
      ss << "Cluster dimensions: " << x << " " << y << " " << z;
```